src/affine.rs

```diff
@@ -114,6 +114,72 @@ impl Affine {
         let p11 = self * Point::new(rect.x1, rect.y1);
         Rect::from_points(p00, p01).union(Rect::from_points(p10, p11))
     }
+
+    /// Compute the singular value decomposition of the linear transformation (ignoring the
+    /// translation).
+    ///
+    /// All non-degenerate linear transformations can be represented as
+    ///
+    ///  1. a rotation about the origin.
+    ///  2. a scaling along the x and y axes
+    ///  3. another rotation about the origin
+    ///
+    /// composed together. Decomposing a 2x2 matrix in this way is called a "singular value
+    /// decomposition" and is written `U Σ V^T`, where U and V^T are orthogonal (rotations) and Σ
+    /// is a diagonal matrix (a scaling).
+    ///
+    /// Since currently this function is used to calculate ellipse radii and rotation from an
+    /// affine map on the unit circle, we don't calculate V^T, since a rotation of the unit (or
+    /// any) circle about its center always results in the same circle. This is the reason that an
+    /// ellipse mapped using an affine map is always an ellipse.
+    ///
+    /// Will return NaNs if the matrix (or equivalently the linear map) is singular.
+    ///
+    /// First part of the return tuple is the scaling, second part is the angle of rotation (in
+    /// radians)
+    #[inline(always)]
+    pub(crate) fn svd(self) -> (Vec2, f64) {
```

Comment:
If this were public API I would suggest using a struct with named fields instead of a tuple as the result type, but this is fine for internal code. 👌

Reply:
Keeping it private for now is good because it does not prevent any public API to be added later.

Reply:
Agree it should be private, but Colin's separate point is that a struct with named fields would be more readable than a tuple.

Reply:
yea sorry if my point was confusing; I think it's good that this is private, but if it *weren't* I would prefer a struct with named fields to a tuple. I think this is fine as is!

Reply:
I actually tried creating a `Svd2x2` struct for the result, but I got lost down a rabbit hole.